doc/designs/passing-algorithmidentifier-parameters.md

```diff
@@ -0,0 +1,156 @@
+Passing AgorithmIdentifier parameters to operations
+===================================================
+
+Quick background
+----------------
+
+We currently only support passing the AlgorithmIdentifier (`X509_ALGOR`)
+parameter field to symmetric cipher provider implementations.
+
+We do support passing them to legacy implementations of other types of
+operation algorithms as well, but it's done in a way that can't be supported
+with providers, because it involves sharing specific structures between
+libcrypto and the backend implementation.
+
+For a longer background and explanation, see
+[Background / tl;dr](#background-tldr) at the end of this design.
+
+Establish an OSSL_PARAM key that any algorithms may become aware of
+-------------------------------------------------------------------
+
+We already have a parameter key, but it's currently only specified for
+`EVP_CIPHER`, in support of `EVP_CIPHER_param_to_asn1()` and
+`EVP_CIPHER_asn1_to_param()`.
+
+"alg\_id\_param", also known as the macro `OSSL_CIPHER_PARAM_ALGORITHM_ID_PARAMS`
```

Comment:
Why not use:
```
`"alg_id_param"`, also known...
```

Reply:
'cause some markdown interpreters interpret in-word underscores as italics markers too, so rendering that like this: alg*id*param .  See https://www.markdownguide.org/basic-syntax/#italic-best-practices for a note on the topic.
I keep forgetting how github markdown behaves...